onnx/serialization.py

```diff
@@ -165,8 +167,43 @@ def deserialize_proto(self, serialized: bytes | str, proto: _Proto) -> _Proto:
         return google.protobuf.json_format.Parse(serialized, proto)
 
 
+class _TextualSerializer(ProtoSerializer):
+    """Serialize and deserialize the ONNX textual representation."""
+
+    supported_format = "onnxtxt"
+    file_extensions = frozenset({".onnxtxt"})
+
+    def serialize_proto(self, proto: _Proto) -> bytes:
+        text = onnx.printer.to_text(proto)  # type: ignore[arg-type]
+        return text.encode(_ENCODING)
+
+    def deserialize_proto(self, serialized: bytes | str, proto: _Proto) -> _Proto:
+        warnings.warn(
+            "The onnxtxt format is experimental. Please report any errors to the ONNX GitHub repository.",
```

Comment:
nit: perhaps mention known limitations as warnings directly (if not many)? Or, once we have the documentation talking about known limitations, we can then provide the document link here.

Reply:
I think we should just link it. For example one of the bugs is being fixed in https://github.com/onnx/onnx/issues/5102